agent/checks/grpc_test.go

```diff
@@ -171,3 +171,12 @@ func TestGRPC_NotProxied(t *testing.T) {
 		}
 	})
 }
+
```

Comment:
Thanks for adding this benchmark test. It's super helpful. 
Could you please add more functional tests? Potential Error Cases to test for could be:
- Create a test that passes invalid options to trigger error case. Confirm logging occurs and errors are exposed.
- Test for dial failure of the endpoint that is being healthchecked, that then succeeds after.